pkg/integration/handler.go

```diff
@@ -0,0 +1,214 @@
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/dhis2-sre/im-manager/pkg/config"
+
+	"github.com/dhis2-sre/im-manager/internal/handler"
+	"github.com/gin-gonic/gin"
+)
+
+func NewHandler(config config.Config, client DockerHubClient) Handler {
+	return Handler{config, client}
+}
+
+type Handler struct {
+	config          config.Config
+	dockerHubClient DockerHubClient
+}
+
+type DockerHubClient interface {
+	GetImages(organization string) ([]string, error)
+	GetTags(organization, repository string) ([]string, error)
+}
+
+type Request struct {
+	Key     string `json:"key" binding:"required"`
+	Payload any    `json:"payload"`
+}
+
+// Integrations ...
+// swagger:route POST /integrations postIntegration
+//
+// Return integration for a given key
+//
+// Security:
+//  oauth2:
+//
+// responses:
+//   200: Any
+//   401: Error
+//   403: Error
+//   415: Error
+func (h Handler) Integrations(c *gin.Context) {
+	var request Request
+	if err := handler.DataBinder(c, &request); err != nil {
+		_ = c.Error(err)
+		return
+	}
+
+	if request.Key == "IMAGE_REPOSITORY" {
+		payload := request.Payload.(map[string]any)
+		organization := payload["organization"].(string)
```

Comment:
I think this can easily lead to a panic https://go.dev/tour/methods/15
If payload["organization"] is not a string.

A bit of error handling will probably give a nicer response to the user. Even though it will probably also work like this as Gin does recover from the panic.

Reply:
org, ok := payload["organization"]
		if !ok {
			_ = c.Error(fmt.Errorf("payload doesn't contain \"organization\""))
			return
		}

		organization, ok := org.(string)
		if !ok {
			_ = c.Error(fmt.Errorf("\"organization\" must be a string"))
			return
		}

Would the above be adequate?